src/main/kotlin/autolotto/winningpoint/WinningRank.kt

```diff
@@ -1,17 +1,27 @@
 package autolotto.winningpoint
 
-enum class WinningRank(val matchingCount: Int, val winningPrice: Long) {
-    FIRST(6, 2000000000),
-    SECOND(5, 30000000),
-    THIRD(4, 1500000),
-    FOURTH(3, 50000),
+enum class WinningRank(val matchingCount: Int, val winningPrice: Long, val isBonusNumberNeeded: Boolean = false) {
+    FIRST(6, 2_000_000_000),
+    SECOND(5, 30_000_000, true),
+    THIRD(5, 1_500_000),
+    FOURTH(4, 50_000),
+    FIFTH(3, 5_000),
     NOTHING(0, 0)
     ;
 
     companion object {
-        fun of(matchingCount: Int): WinningRank {
-            return values().find { it.matchingCount == matchingCount }
-                ?: NOTHING
+        private val winningRanksByCount: Map<Int, List<WinningRank>> = values().groupBy { it.matchingCount }
+
+        fun of(matchingCount: Int, isMatchedBonusNumber: Boolean): WinningRank {
+            if (matchingCount == 5) {
+                return winningRanksByCount[matchingCount]?.find { it.isBonusNumberNeeded == isMatchedBonusNumber }
+                    ?: NOTHING
+            }
+            return winningRanksByCount[matchingCount]?.find { it.matchingCount == matchingCount } ?: NOTHING
+        }
```

Comment:
p3; 이 부분에 대해 조금 고민을 해보았는데, 단순히 boolean만 가지고 있는 게 아니라 predicate 형태로 값을 가지고 있다면 하나로 줄여볼 수 있지 않을까 싶었어요.
단순히 취향 차이일 거 같으니, 다른 방법도 고민해 보시면 좋을 거 같아요. 👍🏽

```suggestion
        fun of(matchingCount: Int, isMatchedBonusNumber: Boolean): WinningRank = winningRanksByCount[matchingCount]
                ?.find { it.isBonusNumberNeeded(isMatchedBonusNumber) }
                ?: NOTHING
```